Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair_degeneracies.h

```diff
@@ -1696,7 +1696,9 @@ bool remove_degenerate_faces(const FaceRange& face_range,
   {
     std::map<edge_descriptor, bool> are_degenerate_edges;
 
-    for(face_descriptor fd : degenerate_face_set)
+    // use a copy of the set to avoid iterator invalidation
+    std::set<face_descriptor> degenerate_face_set_copy;
+    for(face_descriptor fd : degenerate_face_set_copy)
```

Comment:
Problem: `degenerate_face_set_copy` is created empty, and then the loop is on an empty set.

Reply:
Sorry, I of course meant `std::set<face_descriptor> degenerate_face_set_copy(degenerate_face_set)`.

I've now amended my commit.